### PR TITLE
FE: User Management - add initial table

### PR DESCRIPTION
**Pull request by faizanjanjua**
## Description
Description

In the new Settings page, User Management tab, add the initial User Management components:

title and text

Add User button, doesn't do anything yet (just console.log output is fine)

User Management table with mock data for now. The data should be stored in the store. Edit, Resend Email and Revoke Access just have a console.log output for now

Add pagination, client side for now

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2621

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: ui/src/components/Settings/UserManagement.vue [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+            <div class="flex justify-content-between">
+              <h3>User Management</h3>
+              <FeatherButton primary @click="handleAddUser">Add User</FeatherButton>
+            </div>
+          </div>
+          <p>Manage User Access to this instance, including adding, editing, and deleting access.</p>
+        </div>
+      </div>
+      <h2 class="heading_inside p-2">User Management</h2>
+      <div class="container card table-card card border">
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:17]: Remove `card` css class. It's assigned twice

File: ui/src/components/Settings/UserManagement.vue [hunk 2]
```diff
@@ -0,0 +1,209 @@
+<template>
+<section class="container ">
+
+  <TableCard class="card border user-management-wrapper">
+      <div class="header">
+        <div class="title-container">
+          <div class="title-main">
+            <div class="flex justify-content-between">
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:8]: Create 1 class `heading` and add both `  display: flex;` and `  justify-content: space-between;` in it

File: ui/src/components/Settings/UserManagement.vue [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+@use '@featherds/styles/themes/variables';
+@use '@featherds/styles/mixins/typography';
+@use '@featherds/table/scss/table';
+@use '@/styles/_transitionDataTable';
+@use '@/styles/vars.scss';
+
+.flex{
+  display: flex;
+}
+.justify-content-between{
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:134]: Merge both `.flex` and `.justify-content-between` in 1 class name `.heading`

File: ui/src/components/Settings/UserManagement.vue [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+          <div class="title-main">
+            <div class="flex justify-content-between">
+              <h3>User Management</h3>
+              <FeatherButton primary @click="handleAddUser">Add User</FeatherButton>
+            </div>
+          </div>
+          <p>Manage User Access to this instance, including adding, editing, and deleting access.</p>
+        </div>
+      </div>
+      <h2 class="heading_inside p-2">User Management</h2>
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:16]: Remove `p-2` here. Padding already exit in `.heading_inside`. And rename `.heading_inside` to `.heading-inside`

File: ui/src/components/Settings/UserManagement.vue [hunk 5]
```diff
@@ -0,0 +1,209 @@
+<template>
+<section class="container ">
+
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:3]: Remove extra space

File: ui/src/components/Settings/UserManagement.vue [hunk 6]
```diff
@@ -0,0 +1,209 @@
+<template>
+<section class="container ">
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:2]: Remove extra space

File: ui/src/components/Settings/UserManagement.vue [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+  .header{
+      display: flex;
+      justify-content: space-between;
+      padding:0 26px; 
+      .title-container{
+        width: 100%;
+      }
+  }
+  .card {
+    height: 442px;
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:157]: Remove static `height`

File: ui/src/components/Settings/UserManagement.vue [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+        background: var(variables.$surface);
+        text-transform: uppercase;
+      }
+      td {
+        white-space: nowrap;
+        .my-primary-icon {
+          font-size: 1.5rem;
+          color: var(variables.$primary);
+        }
+        div {
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:201]: Remove this `div` css here if it's not beign used.

File: ui/src/components/Settings/UserManagement.vue [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+import { useSettingsStore } from '@/store/Views/settingStore'
+import { SORT } from '@featherds/table'
+import DownloadFile from '@featherds/icon/action/DownloadFile'
+import Refresh from '@featherds/icon/navigation/Refresh'
+import Edit from '@featherds/icon/action/Edit'
+import Email from '@featherds/icon/action/Email'
+import Remove from '@featherds/icon/action/Remove'
+const store = useSettingsStore()
+
+const icons = markRaw({
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:77]: remove unused variable also associated imports

File: ui/src/containers/Settings.vue [hunk 10]
```diff
@@ -13,14 +13,17 @@
...
+        <UserManagement />
+      </FeatherTabPanel>
       <FeatherTabPanel>Third Party Integrations</FeatherTabPanel>
     </FeatherTabContainer>
   </section>
   </div>
 </template>
 
 <script lang="ts" setup>
+import UserManagement from "@/components/Settings/UserManagement.vue";
```
Comment (Shahbaz-dataq) [inline ui/src/containers/Settings.vue:26]: Remove this import we don't need it.

**Review by Shahbaz-dataq — CHANGES_REQUESTED** [review]
Make these changes and run `yarn lint:fix` for lint fixes

File: ui/src/components/Settings/UserManagement.vue [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+  Email,
+  Remove
+})
+
+const columns = [
+  { id: 'name', label: 'Name' },
+]
+
+const unSortedColumns =[
+
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:90]: remove extra space

File: ui/components.d.ts [hunk 12]
```diff
@@ -161,6 +161,7 @@ declare module 'vue' {
     TopologyRightDrawer: typeof import('./src/components/Topology/TopologyRightDrawer.vue')['default']
     TopologySearch: typeof import('./src/components/Topology/TopologySearch.vue')['default']
     ViewSelect: typeof import('./src/components/Topology/ViewSelect.vue')['default']
+    UserManagement: typeof import('./src/components/Settings/UserManagement.vue')['default']
```
[the remark this answers is not in the document]
Comment (faizanjanjua) [reply ui/components.d.ts:164]: done

File: ui/src/components/Settings/UserManagement.vue (hunk 2 above)
Comment (faizanjanjua) [reply ui/src/components/Settings/UserManagement.vue]: resolved

File: ui/src/components/Settings/UserManagement.vue (hunk 11 above)
Comment (faizanjanjua) [reply ui/src/components/Settings/UserManagement.vue]: done

File: ui/src/components/Settings/UserManagement.vue [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+  page: 1, // FE pagination component has base 1 (first page)
+  pageSize: 10,
+  total: 0
+})
+
+const columns = [
+  { id: 'name', label: 'Name' }
+]
+
+const unSortedColumns = [
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:83]: Remove this variable

File: ui/src/components/Settings/UserManagement.vue [hunk 14]
```diff
@@ -0,0 +1,207 @@
...
+                v-for="col of columns"
+                :key="col.label"
+                scope="col"
+                :property="col.id"
+                :sort="(sort as any)[col.id]"
+                v-on:sort-changed="sortChanged"
+              >
+                {{ col.label }}
+              </FeatherSortHeader>
+              <th v-for="col of unSortedColumns"
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:33]: Add simple `th` tag other unSorted columns

**Review by Shahbaz-dataq — CHANGES_REQUESTED** [review]
Just this small change. Pagination logic also needs some work but we'll work on it when we have real data.

File: ui/src/components/Settings/UserManagement.vue [hunk 15]
```diff
@@ -0,0 +1,207 @@
...
+onMounted(async () => await store.getUsersList())
+</script>
+<style lang="scss" scoped>
+@use '@featherds/styles/themes/variables';
+@use '@featherds/styles/mixins/typography';
+@use '@featherds/table/scss/table';
+@use '@/styles/_transitionDataTable';
+@use '@/styles/vars.scss';
+
+.flex{
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:132]: Remove this css class. It's not being used

File: ui/src/components/Settings/UserManagement.vue [hunk 16]
```diff
@@ -0,0 +1,207 @@
...
+  display: flex;
+}
+.heading{
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:135]: Move this `.heading` inside `.user-management-wrapper > .header > .title-container`

File: ui/src/components/Settings/UserManagement.vue [hunk 17]
```diff
@@ -0,0 +1,207 @@
...
+  display: flex;
+  justify-content: space-between;
+}
+.heading-inside{
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:139]: Same move this `.heading-inside` inside `.user-management-wrapper`

**Review by Shahbaz-dataq — CHANGES_REQUESTED** [review]

**Review by Shahbaz-dataq — COMMENTED** [review]
LGTM

File: ui/src/components/Settings/UserManagement.vue [hunk 18]
```diff
@@ -0,0 +1,195 @@
...
+              <th> Resend E-mail </th>
+              <th> Revoke Access </th>
+            </tr>
+          </thead>
+          <TransitionGroup
+            name="data-table"
+            tag="tbody"
+          >
+            <tr v-for="data of store.usersList" v-bind:key="data.id">
+              <td>{{ data.firstName }}</td>
```
Comment (synqotik) [inline ui/src/components/Settings/UserManagement.vue:44]: Should be  `<td>{{ data.firstName }} {{ data.lastName }}</td>`

**Review by synqotik — CHANGES_REQUESTED** [review]
Looks good, just one comment.

However, wait until this PR is done: https://github.com/OpenNMS-Cloud/lokahi/pull/1974

Then rebase onto that to get the changes in that PR for getting user list from BFF.

File: ui/src/components/Settings/UserManagement.vue [hunk 19]
```diff
@@ -0,0 +1,195 @@
...
+              <th> Resend E-mail </th>
+              <th> Revoke Access </th>
+            </tr>
+          </thead>
+          <TransitionGroup
+            name="data-table"
+            tag="tbody"
+          >
+            <tr v-for="data of store.usersList" v-bind:key="data.id">
+              <td>{{ data.firstName }} {{ data.firstName }}</td>
```
Comment (Shahbaz-dataq) [inline ui/src/components/Settings/UserManagement.vue:44]: change the last `firstName` to `lastName`

**Review by Shahbaz-dataq — APPROVED** [review]
Approve. just this small change

**Review by synqotik — APPROVED** [review]
LGTM

**Pull request merged**